html/pfappserver/lib/pfappserver/PacketFence/Controller/Config/Switch.pm

```diff
@@ -274,6 +275,88 @@ sub invalidate_cache :Chained('object') :PathPart('invalidate_cache') :Args(0) {
     $c->response->status(200);
 }
 
+=head1 import_csv
+
+A method to be able to import switches from a CSV
+
+=cut
+
+sub import_csv :Local :Args(0) :AdminRole('SWITCHES_CREATE') {
+    my ( $self, $c ) = @_;
+   
+    my $logger = pf::log->get_logger();
+    my $upload = $c->req->upload('importcsv');
+    my $file = $upload->fh;
+    my $model = $c->model("Config::Switch");
+    my $delimiter = $c->req->param('delimiter');
+    my $model_group = $c->model("Config::SwitchGroup");
+
+    # Map delimiter to its actual character
+    if ($delimiter eq 'comma') {
+        $delimiter = ',';
+    } elsif ($delimiter eq 'semicolon') {
+        $delimiter = ';';
+    } elsif ($delimiter eq 'colon') {
+        $delimiter = ':';
+    } elsif ($delimiter eq 'tab') {
+        $delimiter = "\t";
+    }
+
+    my $skip = 0;
+    my $skip1 = 0;
+    my $switches = 0;
+    my %seen;
+    my $csv = Text::CSV->new({ binary => 1, sep_char => $delimiter });
+    while (my $fields = $csv->getline($file)) {
+
+        unless($skip1) {
+            $skip1 = 1;
+            next;
+        }
+
+        if (@$fields < 3) {
+            $skip++;
+            $logger->warn("This entry has been skipped because this line: @$fields[$_] contains more fields than required");
```

Comment:
@$fields[$_]  is an undefined value.
Would better use the line count.
"This entry has been skipped because this line $line_count contains more fields than required"